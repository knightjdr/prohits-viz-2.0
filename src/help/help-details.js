import React from 'react';

import './help-details.css';

const HelpDetails = () => (
  <div className="HelpDetails-container">
    details
  </div>
);

export default HelpDetails;
