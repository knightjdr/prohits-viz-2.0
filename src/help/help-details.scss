.HelpDetails-container {
  max-height: calc(100vh - 40px);
  padding: 5px 10px;
};
