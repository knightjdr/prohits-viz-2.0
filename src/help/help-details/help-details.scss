.HelpDetails-container {
  flex-grow: 1;
  max-height: calc(100vh - 40px);
  padding: 10px 10px;
};
.HelpDetails-pages {
  height: calc(100vh - 135px);
}
