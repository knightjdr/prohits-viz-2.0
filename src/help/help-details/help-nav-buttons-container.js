import React, { Component } from 'react';

import HelpNavButtons from './help-nav-buttons';

class HelpNavButtonsContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
      length: 3,
    };
  }
  render() {
    return (
      <HelpNavButtons
        index={this.state.index}
        length={this.state.length}
      />
    );
  }
}
export default HelpNavButtonsContainer;
