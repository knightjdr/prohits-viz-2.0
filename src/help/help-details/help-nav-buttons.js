import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faChevronLeft from '@fortawesome/fontawesome-pro-solid/faChevronLeft';
import faChevronRight from '@fortawesome/fontawesome-pro-solid/faChevronRight';
import PropTypes from 'prop-types';
import React from 'react';
import { Button, Icon } from 'antd';

import './help-nav-buttons.css';

const HelpNavButtons = ({
  index,
  length,
}) => (
  <div className="HelpNavButtons-container">
    {
      index > 0 &&
      <Button
        className="HelpNavButtons-backward"
        type="primary"
      >
        <Icon>
          <FontAwesomeIcon icon={faChevronLeft} />
        </Icon>
        Backward
      </Button>
    }
    {
      index < length &&
      <Button
        className="HelpNavButtons-forward"
        type="primary"
      >
        Forward
        <Icon>
          <FontAwesomeIcon icon={faChevronRight} />
        </Icon>
      </Button>
    }
  </div>
);

HelpNavButtons.propTypes = {
  index: PropTypes.number.isRequired,
  length: PropTypes.number.isRequired,
};

export default HelpNavButtons;
