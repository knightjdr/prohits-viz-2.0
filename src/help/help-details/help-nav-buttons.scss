.HelpNavButtons-forward {
  float: right;
}
.HelpNavButtons-container {
  margin-top: 20px;
}
@media screen and (max-width: 680px) {
  .HelpNavButtons-backward {
    margin-left: 40px;
  }
}
