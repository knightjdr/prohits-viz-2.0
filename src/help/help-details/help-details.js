import PropTypes from 'prop-types';
import React from 'react';
import { Breadcrumb } from 'antd';
import { NavLink, Route, Switch, withRouter } from 'react-router-dom';
import RouteNotFound from '../../router/route-not-found';

import HelpNavButtons from './help-nav-buttons-container';
import HelpPages from '../help-pages/help-pages';
import HelpRoutesMap from '../help-routes/help-routes-map';
import RoutesFromPath from '../../helpers/routes-from-path';

import './help-details.css';

export const toRouteNodes = (routes = HelpPages) => {
  let routeArr = [];
  routes.forEach((route) => {
    routeArr.push(
      <Route
        exact
        key={route.route}
        path={route.route}
        component={route.component}
      />,
    );
    if (route.children) {
      routeArr = routeArr.concat(toRouteNodes(route.children));
    }
  });
  return routeArr;
};

export const HelpDetailsComponent = ({
  location,
}) => {
  const breadcrumbElement = RoutesFromPath(location.pathname).map(path => (
    <Breadcrumb.Item
      key={path}
    >
      <NavLink
        to={path}
      >
        { HelpRoutesMap.routeToText[path] }
      </NavLink>
    </Breadcrumb.Item>
  ));
  const contentElement = toRouteNodes();
  return (
    <div className="help__details">
      <div className="help__details-inner">
        <Breadcrumb separator=">">
          { breadcrumbElement }
        </Breadcrumb>
        <div className="help__details-pages">
          <Switch>
            { contentElement }
            <RouteNotFound />
          </Switch>
          <HelpNavButtons />
        </div>
      </div>
    </div>
  );
};

HelpDetailsComponent.propTypes = {
  location: PropTypes.shape({
    path: PropTypes.string,
  }).isRequired,
};

export default withRouter(HelpDetailsComponent);
