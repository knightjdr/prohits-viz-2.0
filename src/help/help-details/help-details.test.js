import React from 'react';
import { shallow } from 'enzyme';

import RoutesFromPath from '../../helpers/routes-from-path';
import { HelpDetails } from './help-details';

// mock HelpRoutes
jest.mock('../help-routes/help-routes');

// mock RoutesFromPath
jest.mock('../../helpers/routes-from-path');
RoutesFromPath.mockReturnValue(['help']);

describe('HelpDetails', () => {
  test('Renders', () => {
    const wrapper = shallow(
      <HelpDetails
        location={{
          path: '/help',
        }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
