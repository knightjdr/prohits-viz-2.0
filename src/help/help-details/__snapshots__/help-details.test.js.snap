// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpDetails Renders 1`] = `
<div
  className="HelpDetails-container"
>
  <Breadcrumb
    prefixCls="ant-breadcrumb"
    separator=">"
  >
    <BreadcrumbItem
      key="help"
      prefixCls="ant-breadcrumb"
      separator="/"
    >
      <Component
        to="help"
      />
    </BreadcrumbItem>
  </Breadcrumb>
  <div
    className="HelpDetails-pages"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        key="/help"
        path="/help"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/help/tools"
        path="/help/tools"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/help/tools/dotplot"
        path="/help/tools/dotplot"
      />
      <Route
        component={[Function]}
        exact={true}
        key="/help/visualization"
        path="/help/visualization"
      />
      <RouteNotFound />
    </Switch>
  </div>
  <HelpNavButtonsContainer />
</div>
`;
